src/components/authentication/InvalidSessionHandler.tsx

```diff
@@ -0,0 +1,33 @@
+import { useTranslations } from 'use-intl';
+import { useEffect, useState } from 'react';
+import { Session } from 'next-auth';
+import { useNotificationSpawner } from 'lib/hooks/UseNotificationSpawner';
+import { useSession } from 'next-auth/react';
+
+/**
+ * Registers a handler for invalid sessions.
+ *
+ * When logout is triggered by an invalid session, this handler will show a notification.
+ */
+export function InvalidSessionHandler() {
+    const [prefSession, setPrefSession] = useState<Session | null>(null);
```

Comment:
You dont need the explicit null in there. This will do exactly the same I think.
```suggestion
    const [prefSession, setPrefSession] = useState<Session>();
```

Reply:
nvm it will be undefined in that case...